modules/secret_manager/main.tf

```diff
@@ -0,0 +1,49 @@
+resource "random_password" "random_pass" {
+  length  = 16
+  special = false
+}
+
+# Recover the GCP secret payload when GCP secret name is provided
+data "google_secret_manager_secret_version" "gcp_predefined_pass" {
+  secret    = var.secret_value
```

Comment:
I think that here you need to use the `secret ID` (or name) and not the payload.

Reply:
You're right, actually I'm reworking at the overhall logic cause I found out a terraform limitation about data retrieving